programs/drift/src/state/fulfillment_params/phoenix.rs

```diff
@@ -0,0 +1,481 @@
+        spot_market::{SpotBalanceType, SpotFulfillmentConfigStatus, SpotMarket},
+        state::State,
+        traits::Size,
+    },
+    validate,
+};
+
+pub const PHOENIX_MARKET_DISCRIMINANT: u64 = 8167313896524341111;
+
+#[account(zero_copy)]
+#[derive(Default, PartialEq, Eq, Debug)]
+#[repr(C)]
+pub struct PhoenixV1FulfillmentConfig {
+    pub pubkey: Pubkey,
+    pub phoenix_program_id: Pubkey,
+    pub phoenix_market: Pubkey,
+    pub market_index: u16,
+    pub fulfillment_type: SpotFulfillmentType,
+    pub status: SpotFulfillmentConfigStatus,
+    pub padding: [u8; 4],
+}
+
+impl Size for PhoenixV1FulfillmentConfig {
+    const SIZE: usize = 112;
+}
+
+#[derive(Clone)]
+pub struct PhoenixMarketContext<'a, 'b> {
+    pub phoenix_market: &'a AccountInfo<'b>,
+    pub header: MarketHeader,
+}
+
+impl<'a, 'b> PhoenixMarketContext<'a, 'b> {
+    pub fn new(info: &'a AccountInfo<'b>) -> DriftResult<PhoenixMarketContext<'a, 'b>> {
+        validate!(
+            info.owner == &phoenix::id(),
+            ErrorCode::InvalidPhoenixProgram,
+            "Market must be owned by the Phoenix program",
+        )?;
+        let data = info.data.borrow();
+        let header = bytemuck::try_from_bytes::<MarketHeader>(&data[..size_of::<MarketHeader>()])
+            .map_err(|_| {
+            msg!("Failed to parse Phoenix market header");
+            ErrorCode::FailedToDeserializePhoenixMarket
+        })?;
+        validate!(
+            header.discriminant == PHOENIX_MARKET_DISCRIMINANT,
+            ErrorCode::InvalidPhoenixProgram,
+            "Invalid market discriminant",
+        )?;
+        Ok(PhoenixMarketContext {
+            phoenix_market: info,
+            header: *header,
+        })
+    }
+
+    pub fn to_phoenix_v1_fulfillment_config(
+        &self,
+        config_key: &Pubkey,
+        market_index: u16,
+    ) -> PhoenixV1FulfillmentConfig {
+        PhoenixV1FulfillmentConfig {
+            pubkey: *config_key,
+            phoenix_program_id: phoenix::id(),
+            phoenix_market: *self.phoenix_market.key,
+            market_index,
+            fulfillment_type: SpotFulfillmentType::PhoenixV1,
+            status: SpotFulfillmentConfigStatus::Enabled,
+            padding: [0; 4],
+        }
+    }
+}
+
+impl<'a, 'b> Deref for PhoenixMarketContext<'a, 'b> {
+    type Target = AccountInfo<'b>;
+
+    fn deref(&self) -> &Self::Target {
+        self.phoenix_market
+    }
+}
+
+#[derive(Clone)]
+pub struct PhoenixFulfillmentParams<'a, 'b> {
+    pub phoenix_program: &'a AccountInfo<'b>,
+    pub phoenix_log_authority: &'a AccountInfo<'b>,
+    pub phoenix_market: PhoenixMarketContext<'a, 'b>,
+    pub phoenix_base_vault: &'a AccountInfo<'b>,
+    pub phoenix_quote_vault: &'a AccountInfo<'b>,
+    pub drift_signer: &'a AccountInfo<'b>,
+    pub base_market_vault: Box<Account<'b, TokenAccount>>,
+    pub quote_market_vault: Box<Account<'b, TokenAccount>>,
```

Comment:
My rough understanding is the `drift_signer` is the owner of these 2 token accounts. Not immediately obvious, but wondering if there's a section in the source where this is explicitly documented

Reply:
I think a better semantic name for `drift_signer` might be `drift_vault_authority`. Then if you rename `base_market_vault` to `drift_base_market_vault`, the default ownership can be more intuitively deduced via the variable naming. Not a big deal